sonar-php-plugin/src/main/java/org/sonar/plugins/php/phpunit/PhpUnitCoverageResultParser.java

```diff
@@ -117,114 +115,98 @@ private void parseFile(File coverageReportFile) {
    * @param resolvedPaths list of paths which can be mapped to imported files
    */
-  protected void saveCoverageMeasure(FileNode fileNode, List<String> unresolvedPaths, List<String> resolvedPaths) {
+  protected void saveCoverageMeasure(FileNode fileNode, List<String> unresolvedPaths, List<String> resolvedPaths, SensorContext context) {
     //PHP supports only absolute paths
     String path = fileNode.getName();
     InputFile inputFile = fileSystem.inputFile(fileSystem.predicates().hasAbsolutePath(path));
 
     // Due to an unexpected behaviour in phpunit.coverage.xml containing references to covered source files, we have to check that the
     // targeted file for coverage is not null.
     if (inputFile != null) {
-      org.sonar.api.resources.File phpFile = org.sonar.api.resources.File.create(inputFile.relativePath());
       resolvedPaths.add(inputFile.relativePath());
-      saveCoverageLineHitsData(fileNode, inputFile, phpFile);
-
-      // Save uncovered statements (lines)
-      MetricsNode metrics = fileNode.getMetrics();
-      double totalStatementsCount = metrics.getTotalStatementsCount();
-      double uncoveredLines = totalStatementsCount - metrics.getCoveredStatements();
-      context.saveMeasure(phpFile, linesToCoverMetric, totalStatementsCount);
-      context.saveMeasure(phpFile, this.uncoveredLinesMetric, uncoveredLines);
+      saveCoverageLineHitsData(fileNode, inputFile, context);
+
+      // Saving the uncovered statements (lines) is no longer needed because coverage metrics are internally derived by the NewCoverage
     } else {
       unresolvedPaths.add(path);
     }
   }
 
-  private void saveCoverageLineHitsData(FileNode fileNode, InputFile inputFile, org.sonar.api.resources.File phpFile) {
+  private void saveCoverageLineHitsData(FileNode fileNode, InputFile inputFile, SensorContext context) {
     // Properties builder will generate the data associate with COVERAGE_LINE_HITS_DATA metrics.
     // This should look like (lineNumber=Count) : 1=0;2=1;3=1....
+    NewCoverage newCoverage = context.newCoverage().onFile(inputFile).ofType(coverageType);
     PropertiesBuilder<Integer, Integer> lineHits = new PropertiesBuilder<>(coverageLineHitsDataMetric);
     if (fileNode.getLines() != null) {
       for (LineNode line : fileNode.getLines()) {
         int lineNum = line.getNum();
         if (lineNum > 0 && lineNum <= inputFile.lines()) {
-          saveLineMeasure(line, lineHits);
+          newCoverage.lineHits(line.getNum(), line.getCount());
         } else {
-          LOG.warn(String.format(WRONG_LINE_EXCEPTION_MESSAGE, lineNum, phpFile.getLongName()));
+          LOG.warn(String.format(WRONG_LINE_EXCEPTION_MESSAGE, lineNum, inputFile.file()));
         }
       }
     }
-    Measure measure = lineHits.build();
-    logMeasureByResource(phpFile, measure);
-    context.saveMeasure(phpFile, measure);
+
+    Measure<String> measure = lineHits.build();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "Measure"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)